tests/test_macros.cpp

```diff
@@ -0,0 +1,28 @@
+#include <mdspan/mdspan.hpp>
+#include <gtest/gtest.h>
+
+TEST(mdspan_macros, precondition_violation)
+{
+#if defined(_MDSPAN_HAS_CUDA) || defined(_MDSPAN_HAS_HIP) || defined(_MDSPAN_HAS_SYCL)
+  constexpr auto msg = "";
+#else
+  constexpr auto msg = "hello, world!";
+#endif
+
```

Comment:
I would have done
```c++
#if MDSPAN_IMPL_CHECK_PRECONDITION
ASSERT_DEATH(MDSPAN_IMPL_PRECONDITION(false and "hello, world!"), msg);
#else
ASSERT_NO_THROW(MDSPAN_IMPL_PRECONDITION(false and "hello, world!"));
#endif
```
instead of messing with the test success criterion in the cmake file.

Reply:
Please ignore.   This doesn't apply here.  I is only for `test_precondition_checks_can_be_disabled` Sorry about the noise.